ion-schema/src/nfa.rs

```diff
@@ -156,8 +155,11 @@ impl NfaEvaluation {
     }
 
     /// Validates provided ordered elements against referenced [Nfa]
-    pub fn validate_ordered_elements(&mut self, elements: &[Element], type_store: &TypeStore) {
-        let mut elements_iter = elements.iter().peekable();
+    pub fn validate_ordered_elements<'a, T: Iterator<Item = &'a Element>>(
```

Comment:
🗺️ The NFA implementation now accepts any iterator over `Element`s, rather than requiring a slice of `Element`s or being tied to a particular iterator implementation.